src/i2c.rs

```diff
@@ -852,3 +928,240 @@ where
         }
     }
 }
+
+#[cfg(any(feature = "stm32f413", feature = "stm32f423",))]
+impl<I2C, PINS> FMPI2c<I2C, PINS>
+where
+    I2C: Deref<Target = fmpi2c::RegisterBlock>,
+{
+    fn i2c_init(&self, speed: KiloHertz, pclk: Hertz) {
+        use core::cmp;
+
+        // Make sure the I2C unit is disabled so we can configure it
+        self.i2c.cr1.modify(|_, w| w.pe().clear_bit());
+
+        // Calculate settings for I2C speed modes
+        let presc;
+        let scldel;
+        let sdadel;
+        let sclh;
+        let scll;
+
+        let freq: u32 = pclk.0;
+
+        // Normal I2C speeds use a different scaling than fast mode below
+        if speed <= 100_u32.khz() {
+            presc = 1;
+            scll = cmp::max((((freq >> presc) >> 1) / speed.0) - 1, 255) as u8;
+            sclh = scll - 4;
+            sdadel = 2;
+            scldel = 4;
+        } else {
+            presc = 0;
+            scll = cmp::max((((freq >> presc) >> 1) / speed.0) - 1, 255) as u8;
+            sclh = scll - 6;
+            sdadel = 1;
+            scldel = 3;
+        }
+
+        // Enable I2C signal generator, and configure I2C for 400KHz full speed
```

Comment:
Inconsistent comment?

Reply:
Yes, thanks for noticing. Guess I'll need to fix it in stm32f0xx-hal, too. ;)